plugins/SlicerT/SlicerTWaveform.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * SlicerTWaveform.h - declaration of class SlicerTWaveform
+ *
+ * Copyright (c) 2023 Daniel Kauss Serna <daniel.kauss.serna@gmail.com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef LMMS_SLICERT_WAVEFORM_H
+#define LMMS_SLICERT_WAVEFORM_H
+
+#include <QApplication>
+#include <QFontMetrics>
+#include <QInputDialog>
+#include <QMouseEvent>
+#include <QPainter>
+
+#include "Instrument.h"
+#include "SampleBuffer.h"
+
+namespace lmms {
+
+class SlicerT;
+
+namespace gui {
+
+class SlicerTWaveform : public QWidget
+{
+	Q_OBJECT
+
+public slots:
+	void updateUI();
+	void isPlaying(float current, float start, float end);
+
+public:
+	SlicerTWaveform(int w, int h, SlicerT* instrument, QWidget* parent);
+
+	// predefined sizes
+	static constexpr int m_seekerHorMargin = 5;
+	static constexpr int m_seekerHeight = 38; // used to calcualte all vertical sizes
+	static constexpr int m_middleMargin = 6;
+
+	// colors
+	const QColor s_emptyColor = QColor(0, 0, 0, 0);
+
+	const QColor s_waveformColor = QColor(123, 49, 212);
+	const QColor s_waveformBgColor = QColor(255, 255, 255, 0);
+	const QColor s_waveformMaskColor = QColor(151, 65, 255); // update this if s_waveformColor changes
+	const QColor s_waveformInnerColor = QColor(183, 124, 255);
+
+	const QColor s_playColor = QColor(255, 255, 255, 200);
+	const QColor s_playHighlightColor = QColor(255, 255, 255, 70);
+
+	const QColor s_sliceColor = QColor(218, 193, 255);
+	const QColor s_sliceShadowColor = QColor(136, 120, 158);
+	const QColor s_sliceHighlightColor = QColor(255, 255, 255);
+
+	const QColor s_seekerColor = QColor(178, 115, 255);
+	const QColor s_seekerHighlightColor = QColor(178, 115, 255, 100);
+	const QColor s_seekerShadowColor = QColor(0, 0, 0, 120);
```

Comment:
doesn't work:
`error: ‘constexpr’ needed for in-class initialization of static data member ‘const QColor lmms::gui::SlicerTWaveform::s_seekerShadowColor’ of non-integral type`

And what exactly is wrong with just using `const`? I can't think of anything where it would conflict, although I'm not very familiar with c++.

Reply:
[They prevent copying and moving](https://stackoverflow.com/questions/38155699/are-there-advantages-of-using-const-member-variable-in-c). Try making them only `static` and defining them inside `SlicerTWaveForm`.cpp and not in here.

Reply:
Done I think